x-pack/plugins/enterprise_search/server/routes/enterprise_search/crawler/crawler.ts

```diff
@@ -35,55 +35,75 @@ export function registerCrawlerRoutes(routeDependencies: RouteDependencies) {
-            'xpack.enterpriseSearch.server.routes.addCrawler.crawlerExistsError',
-            {
-              defaultMessage: 'A crawler for this index already exists',
-            }
-          ),
-          response,
-          statusCode: 409,
+
+      try {
+        const indexExists = await client.asCurrentUser.indices.exists({
+          index: request.body.index_name,
         });
-      }
+        if (indexExists) {
+          return createError({
+            errorCode: ErrorCode.INDEX_ALREADY_EXISTS,
+            message: i18n.translate(
+              'xpack.enterpriseSearch.server.routes.addCrawler.indexExistsError',
+              {
+                defaultMessage: 'This index already exists',
+              }
+            ),
+            response,
+            statusCode: 409,
+          });
+        }
+        const crawler = await fetchCrawlerByIndexName(client, request.body.index_name);
+        if (crawler) {
+          return createError({
+            errorCode: ErrorCode.CRAWLER_ALREADY_EXISTS,
+            message: i18n.translate(
+              'xpack.enterpriseSearch.server.routes.addCrawler.crawlerExistsError',
+              {
+                defaultMessage: 'A crawler for this index already exists',
+              }
+            ),
+            response,
+            statusCode: 409,
+          });
+        }
 
-      const connector = await fetchConnectorByIndexName(client, request.body.index_name);
+        const connector = await fetchConnectorByIndexName(client, request.body.index_name);
 
-      if (connector) {
-        return createError({
-          errorCode: ErrorCode.CONNECTOR_DOCUMENT_ALREADY_EXISTS,
-          message: i18n.translate(
-            'xpack.enterpriseSearch.server.routes.addCrawler.connectorExistsError',
-            {
-              defaultMessage: 'A connector for this index already exists',
-            }
-          ),
-          response,
-          statusCode: 409,
-        });
+        if (connector) {
+          return createError({
+            errorCode: ErrorCode.CONNECTOR_DOCUMENT_ALREADY_EXISTS,
+            message: i18n.translate(
+              'xpack.enterpriseSearch.server.routes.addCrawler.connectorExistsError',
+              {
+                defaultMessage: 'A connector for this index already exists',
+              }
+            ),
+            response,
+            statusCode: 409,
+          });
+        }
+
+        return enterpriseSearchRequestHandler.createRequest({
+          path: '/api/ent/v1/internal/indices',
+        })(context, request, response);
+      } catch (e) {
+        // 403 UNAUTHORIZED
+        if (e.meta.statusCode === 403) {
+          return createError({
+            errorCode: ErrorCode.UNAUTHORIZED,
+            message: i18n.translate(
+              'xpack.enterpriseSearch.server.routes.addCrawler.unauthorizedError',
+              {
+                defaultMessage: 'This account does not have permission to do this.',
```

Comment:
@serenachou can you good word better me here?

Reply:
"You do not have sufficient permissions to create index ${request.body.index_name}"

Reply:
Also, might be best to create an error messages file where we reuse this error message? Not sure.

Reply:
Instead of "you" can we actually specify the username that the user is signed in as? ${username} ?

Reply:
i would love a error messages file for easier updates with @leemthompo  and @chriscressman !